classloader-leak-prevention/classloader-leak-prevention-core/src/test/java/se/jiderhamn/classloader/leak/prevention/cleanup/SAAJEnvelopeFactoryParserPoolCleanUpTest.java

```diff
@@ -28,26 +25,45 @@ protected void triggerLeak() throws Exception {
     }
     */
 
+    Class<?> envelopeFactoryClass = preventor.findClass("com.sun.xml.internal.messaging.saaj.soap.EnvelopeFactory");
+    if (envelopeFactoryClass == null) {
+      // Try the package for the maven dependency if the internal one is not available
+      envelopeFactoryClass = preventor.findClass("com.sun.xml.messaging.saaj.soap.EnvelopeFactory");
+    }
+    
     final Object /* com.sun.xml.internal.messaging.saaj.soap.ContextClassloaderLocal */
-            parserPool = preventor.getStaticFieldValue(EnvelopeFactory.class, "parserPool");
-    final ParserPool currentParserPool = (ParserPool) preventor.findMethod(parserPool.getClass().getSuperclass(), "get").invoke(parserPool);
+            parserPool = preventor.getStaticFieldValue(envelopeFactoryClass, "parserPool");
+    final Object currentParserPool = preventor.findMethod(parserPool.getClass().getSuperclass(), "get").invoke(parserPool);
     assertNotNull(currentParserPool);
 
-    final SAXParser saxParser = currentParserPool.get();
-    saxParser.setProperty("http://apache.org/xml/properties/internal/error-handler", 
-        new CustomErrorHandler()); // Loaded by protected classloader
-    currentParserPool.put(saxParser);
+    final Method getMethod = preventor.findMethod(currentParserPool.getClass(), "get");
+    final SAXParser saxParser = (SAXParser) getMethod.invoke(currentParserPool);
+    
+    saxParser.setProperty("http://apache.org/xml/properties/internal/error-handler", getCustomErrorHandlerInstance(preventor));
+    
+    final Method putMethod = preventor.findMethod(currentParserPool.getClass(), "put", SAXParser.class);
+    putMethod.invoke(currentParserPool, saxParser);
   }
 
-  /** Dummy XMLErrorHandler loaded by the class loader that should be garbage collected*/
-  public static class CustomErrorHandler implements com.sun.org.apache.xerces.internal.xni.parser.XMLErrorHandler {
-    @Override
-    public void warning(String domain, String key, XMLParseException exception) throws XNIException { }
-
-    @Override
-    public void error(String domain, String key, XMLParseException exception) throws XNIException { }
+  /*
+   * Create a dummy XMLErrorHandler to be loaded by the classloader that sould be garbage collected
+   * Create using reflection, since the type is not accessible in newer Java Versions
+   */
+  public Object getCustomErrorHandlerInstance(final ClassLoaderLeakPreventor preventor) {
+    final Class<?> errorHandlerClass = preventor.findClass("com.sun.org.apache.xerces.internal.xni.parser.XMLErrorHandler");
```

Comment:
Is this classname the same for both old JDK and Maven dependency? In that case, why the switch to reflection...?

Reply:
The class `com.sun.org.apache.xerces.internal.xni.parser.XMLErrorHandler` is still present in the JDK, we do not have to add it back with an external dependency. We have to use reflection, because the internal sun classes can not be accessed directly anymore, and the class will not compile with JDK 11.

Reply:
Ah, yes. I'd prefer to have a comment about in the code explaining that.

Reply:
I see. I've tried clarifying the comment (that I misread).